src/app/archive_hardfork_toolbox/sql.ml

```diff
@@ -1,8 +1,13 @@
+open Async
 open Core
 open Caqti_request.Infix
 
 module type CONNECTION = Mina_caqti.CONNECTION
 
+type genesis_block = { id : int; state_hash : string }
+
+type block_info = { id : int; height : int64; protocol_version_id : int }
```

Comment:
```sql
CREATE TABLE protocol_versions
( id               serial PRIMARY KEY
, transaction      int    NOT NULL
, network          int    NOT NULL
, patch            int    NOT NULL
, UNIQUE (transaction,network,patch)
);
```

protocol_version_id is just the primary key in this table. Since we've confirmed that in previous PR, we don't care about the stability of IDs, we should locate protocol version by the triple (transaction, network, patch) instead.